contrib/federations_tutorial/BlockSigning.py

```diff
@@ -0,0 +1,139 @@
+#!/usr/bin/env python
+import threading
+import multiprocessing
+import json
+from kafka import KafkaConsumer, KafkaProducer
+from time import sleep, time
+from authproxy import JSONRPCException
+
+KAFKA_SERVER = 'localhost:9092'
+TOPIC_NEW_BLOCK = 'new-block'
+TOPIC_NEW_SIG = 'new-sig'
+
+class Producer(threading.Thread):
+    def __init__(self, height, block):
+        threading.Thread.__init__(self)
+        self.stop_event = threading.Event()
+        self.daemon = True
+        self.new_height = height + 1
+        self.block = block
+        self.producer = KafkaProducer(bootstrap_servers=KAFKA_SERVER)
+
+    def stop(self):
+        self.stop_event.set()
+
+    def run(self):
+        # send new block proposal
+        self.producer.send(TOPIC_NEW_BLOCK,
+                    key=str.encode('{}'.format(self.new_height)),
+                    value=str.encode(self.block))
+
+        while not self.stop_event.is_set():
+            sleep(1)
+
+        self.producer.close()
+
+class Consumer(threading.Thread):
+    def __init__(self, id, height, elements):
+        threading.Thread.__init__(self)
+        self.stop_event = threading.Event()
+        self.daemon = True
+        self.sig_topic = TOPIC_NEW_SIG + "{}".format(id)
+        self.height = height
+        self.elements = elements
+        self.consumer = KafkaConsumer(bootstrap_servers=KAFKA_SERVER,
+                             auto_offset_reset='earliest',
+                             consumer_timeout_ms=1000)
+
+    def stop(self):
+        self.stop_event.set()
+
+    def run(self):
+        self.consumer.subscribe([TOPIC_NEW_BLOCK])
+
+        # wait for new block proposal and sign block when received
+        while not self.stop_event.is_set():
+            for message in self.consumer:
+                new_height = int(message.key.decode())
+                if new_height > self.height: # just in case to avoid old messages
+                    new_block = message.value.decode()
+                    try:
+                        sign = self.elements.signblock(new_block)
+                        reply = {'key': new_height, 'sig': sign}
+                        producer = KafkaProducer(bootstrap_servers=KAFKA_SERVER,
+                                                 value_serializer=lambda v: json.dumps(v).encode('utf-8'))
+                        producer.send(self.sig_topic, reply)
+                        producer.close()
+                    except JSONRPCException as error:
+                        print(error)
+
+            if self.stop_event.is_set():
```

Comment:
This seems unnecessary, the while loop will stop at the same point without these 2 lines, no?